components/api_request_helper/api_request_helper.cc

```diff
@@ -335,106 +314,177 @@ void APIRequestHelper::URLLoaderHandler::send_sse_data_for_testing(
   OnDataReceived(string_piece, base::BindOnce([]() {}));
 }
 
+data_decoder::DataDecoder*
+APIRequestHelper::URLLoaderHandler::GetDataDecoder() {
+  if (!data_decoder_) {
+    VLOG(1) << "Creating DataDecoder for APIRequestHelper";
+    data_decoder_ = std::make_unique<data_decoder::DataDecoder>();
+  }
+  return data_decoder_.get();
```

Comment:
lazy-load decoder, scoped to a single non-reusable request, just like previous calls to `ParseJsonIsolated`